api/src/main/java/ch/liip/timeforcoffee/api/mappers/ConnectionMapper.java

```diff
@@ -14,9 +14,14 @@ public static Connection fromZvv(ch.liip.timeforcoffee.zvv.CheckPoint checkPoint
         checkPointLocation.setLongitude(checkPoint.getLocation().getLng());
         checkPointLocation.setLatitude(checkPoint.getLocation().getLat());
 
-        Date departureScheduled = checkPoint.getDeparture() == null ? null : checkPoint.getDeparture().getScheduled();
-        Date departureRealtime = checkPoint.getDeparture() == null ? null : checkPoint.getDeparture().getRealtime();
-        Date arrivalScheduled = checkPoint.getArrival() == null ? null : checkPoint.getArrival().getScheduled();
```

Comment:
Are you confident that checkpoint will never have null departure or arrival?

Reply:
I wasn't able to find any case that returns a null departure or arrival but, as somebody added them, I won't remove these checks until we have a new backend.